docs/se/oci/03_vault.adoc

```diff
@@ -0,0 +1,240 @@
+    <artifactId>helidon-integrations-oci-vault</artifactId>
+</dependency>
+----
+
+
+== Setting up the OCI Vault
+
+
+In order to use the OCI Vault integration, the following setup should be made.
+
+* The configuration required for Vault integration includes:
+* Vault OCID - to use the correct Vault, as more than one can be configured
+* Compartment OCID - OCI-specific compartment
+* Encryption Key OCID - required when doing encryption/decryption
+* Signature Key OCID - required when doing signatures/verification
+* Cryptographic endpoint - required for all except secrets
+
+First specify OCIDs and URLs of Vault items in `application.yaml`:
+
+[source,yaml]
+----
+oci:
+  vault:
+    vault-ocid: "<...>"
+    compartment-ocid: "<...>"
+    encryption-key-ocid: "<...>"
+    signature-key-ocid: "<...>"
+    cryptographic-endpoint: "<...>"
+----
+Current configuration requires `~/.oci/config` to be available in the home folder. This configuration file can be downloaded from OCI.
+
+The OCIDs can be set up and found in OCI under Security tab.
+
+image::oci/ocivault.png[OCI Vault, align="center"]
+
+Next, these values should be read and provided to `VaultService`:
+
+[source,java]
+----
+Config vaultConfig = config.get("oci.vault");
+// the following three parameters are required
+String vaultOcid = vaultConfig.get("vault-ocid").asString().get();
+String compartmentOcid = vaultConfig.get("compartment-ocid").asString().get();
+String encryptionKey = vaultConfig.get("encryption-key-ocid").asString().get();
+String signatureKey = vaultConfig.get("signature-key-ocid").asString().get();
+
+// this requires OCI configuration in the usual place
+// ~/.oci/config
+OciVaultRx ociVault = OciVaultRx.create(config.get("oci"));
+
+WebServer.builder()
+        .config(config.get("server"))
+        .routing(Routing.builder()
+                         .register("/vault", new VaultService(ociVault,
+                                                              vaultOcid,
+                                                              compartmentOcid,
+                                                              encryptionKey,
+                                                              signatureKey)))
+        .build()
+        .start()
+        .await();
+----
+
+The `VaultService` should define an `update` method to map paths to handler methods:
+
+[source,java]
+----
+@Override
+public void update(Routing.Rules rules) {
+    rules.get("/encrypt/{text:.*}", this::encrypt)
+            .get("/decrypt/{text:.*}", this::decrypt)
+            .get("/sign/{text}", this::sign)
+            .get("/verify/{text}/{signature:.*}", this::verify)
+            .get("/secret/{id}", this::getSecret)
+            .post("/secret/{name}", Handler.create(String.class, this::createSecret))
+            .delete("/secret/{id}", this::deleteSecret);
+}
+----
+
+== OCI Vault usage
+
+=== Encryption
+To encrypt a text, submitted with a GET request to `/encrypt` endpoint:
```

Comment:
To encrypt a text, submit a GET request to the `/encrypt` endpoint:

Reply:
done